src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -314,7 +314,7 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
             var options = _optionsFactory.Create(ScriptApplicationHostOptionsSetup.SkipPlaceholder);
             RunFromPackageContext pkgContext = assignmentContext.GetRunFromPkgContext();
 
-            if (_environment.SupportsAzureFileShareMount())
+            if (_environment.SupportsAzureFileShareMount() || pkgContext.IsRunFromLocalPackage())
```

Comment:
i think it would help to have more tests to cover all the new logic in this file since there is a lot of complexity here.

recommend listing all the different possible combinations ex: azure files, rfp=1, rfp=external url and failures at different places and add some tests to document the expected behavior